barretenberg/cpp/pil/vm2/bytecode/bc_hashing.pil

```diff
@@ -66,68 +67,168 @@ namespace bc_hashing;
     // to be activated which is impossible on first row.), LATCH_CONDITION is a boolean.
     pol LATCH_CONDITION = latch + precomputed.first_row;
 
-    // The start of a new bytecode id and new set of hashing runs. Needs to be a committed column as it is used in the lookup
-    pol commit start;
-    start * (1 - start) = 0;
+    // The start of a new bytecode id and new set of hashing runs:
+    pol START = sel * (1 - sel_not_start);
+
+    // Needs to be a committed column as it is used in the lookup:
+    pol commit sel_not_start;
+    sel_not_start * (1 - sel_not_start) = 0;
 
     // If the current row is a latch or the first row, the next row should be a start (if it's active)
     #[START_AFTER_LATCH]
-    sel' * (start' - LATCH_CONDITION) = 0;
+    sel' * ((1 - sel_not_start') - LATCH_CONDITION) = 0;
 
     // Used as part of the lookup into bytecode decomposition
```

Comment:
Would it simplify what follows if you could generate the hash of a bytecode as
```
poseidon2([SEPARATOR, 0, 0, field0, field1, field2...]);
```
?

If it simplifies things enough (not sure) it might be possible to consider changing it in the deployer contract?

Reply:
It may, but I was hesistant to change the definition of the hash (at least in this PR) since it would require a lot of changes in the noir contract, protocol contract fixtures, ts, and may brick a load of tests. My logic was that a classic Miranda-sized PR probably wasn't worth simplifying 15 or so lines of PIL code 😅 
However I'd be happy to do a follow up - this would also make it easier to see the impact of 'lengthening' the hash by two fields?